gatsby-node.js

```diff
@@ -1,31 +1,134 @@
+  const typeDefs = `
+    type MedusaProduct implements Node {
+      id: ID!
+      title: String!
+      subtitle: String
+      description: String
+      handle: String!
+      is_giftcard: Boolean!
+      status: String!
+      images: [MedusaImage]
+      thumbnail: String
+      options: [MedusaProductOption]!
+      variants: [MedusaProductVariant]!
+      collection: MedusaCollection
+      collection_id: String
+      profile_id: String!
+      discountable: Boolean!
+      published_at: Date!
+      created_at: Date!
+      updated_at: Date!
+      weight: Int
+      length: Int
+      width: Int
+    }
+    type MedusaCollection @dontInfer {
+      id: ID!
+      handle: String!
+      title: String!
+      created_at: Date!
+      updated_at: Date!
+    }
+    type MedusaProductOption @dontInfer {
+      id: ID!
+      title: String!
+      product_id: String!
+      values: [MedusaProductOptionValue]!
+      created_at: Date!
+      updated_at: Date!
+    }
+    type MedusaProductOptionValue @dontInfer {
+      id: ID!
+      value: String!
+      created_at: Date!
+      updated_at: Date!
+      option_id: String!
+      variant_id: String!
+    }
+    type MedusaProductVariant @dontInfer {
+      id: ID!
+      title: String!
+      product_id: String!
+      prices: [MedusaMoneyAmount]!
+      sku: String
+      barcode: String
+      upc: String
+      variant_rank: Int
+      inventory_quantity: Int!
+      allow_backorder: Boolean!
+      manage_inventory: Boolean!
+      hs_code: String
+      origin_country: String
+      mid_code: String
+      material: String
+      weight: Int
+      length: Int
+      height: Int
+      width: Int
+      options: [MedusaProductOptionValue]!
+      created_at: Date!
+      updated_at: Date!
+    }
+    type MedusaMoneyAmount @dontInfer {
+      id: ID!
+      amount: Int!
+      currency_code: String!
+      created_at: Date!
+      updated_at: Date!
+      variant_id: String!
+    }
+    type MedusaImage @dontInfer {
+      id: ID!
+      url: String!
+      created_at: Date!
+      updated_at: Date!
+    }
+    type MedusaRegion implements Node {
+      id: ID!
+      name: String!
+      tax_rate: Float!
+      currency_code: String!
+      created_at: Date!
+      updated_at: Date!
+      countries: [MedusaCountry]!
+    }
+    type MedusaCountry @dontInfer {
+      id: ID!
+      display_name: String!
+      iso_2: String!
+      iso_3: String!
+      name: String!
+      num_code: Int!
+      region_id: String!
+    }
+  `
+
+  createTypes(typeDefs)
+}
+
 exports.sourceNodes = async ({
   actions: { createNode },
   createContentDigest,
+  reporter,
 }) => {
   const products = await client.products
     .list()
     .then(response => response.products)
+    .catch(_ => {
+      reporter.panic(
+        "Could not fetch products from Medusa, please ensure that you have a Medusa server running and that you have set the correct BASE_URL in the client."
+      )
+      return
```

Comment:
Now that I come to think of it, it might sense to still allow the application to run but without creating any product pages. In the future, we might tweak the landing page to be more of a step-by-step guide on how to get started with Medusa Express. Therefore, disallowing people to open the application might create more harm than good. 

What do you think @kasperkristensen ? :)